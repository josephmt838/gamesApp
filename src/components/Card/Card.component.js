import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

import "./Card.scss";

const Card = ({
  link,
  children,
  handleClick,
  id,
  flipped,
  back,
  front,
  height,
  width,
  title,
  disabled,
  solved
}) => {
  return (
    <Fragment>
      {link && link.length ? (
        <Link to={link} style={{ textDecoration: "none" }}>
          <div className="card">{children}</div>
        </Link>
      ) : (
        <div
          className={`card ${flipped ? "flipped" : ""}`}
          onClick={() => (disabled || solved ? null : handleClick(id))}
        >
          {children ? (
            children
          ) : (
            <div className="flipper">
              <img
                style={{ height, width }}
                src={solved || flipped ? front : back}
                alt="card"
              />
              <h3>{flipped || solved ? title : ""}</h3>
            </div>
          )}
        </div>
      )}
    </Fragment>
  );
};

Card.propTypes = {
  link: PropTypes.string,
  children: PropTypes.node,
  handleClick: PropTypes.func,
  flipped: PropTypes.bool,
  back: PropTypes.string,
  front: PropTypes.string,
  height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  disabled: PropTypes.bool,
  title: PropTypes.string
};

export default Card;
