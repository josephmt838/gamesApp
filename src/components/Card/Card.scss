.card {
  width: 200px;
  height: 200px;
  border: 1px solid lightblue;
  border-radius: 5px;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  color: darkblue;
  margin: 0 auto;
  text-align: center;
  &:hover {
    border-color: lightcoral;
  }
  h3 {
    margin: 0;
  }
}
