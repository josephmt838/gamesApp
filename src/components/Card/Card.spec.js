import React from "react";
import renderer from "react-test-renderer";
import Card from "../Card";

describe("Card Component", () => {
  it("Renders Card component", () => {
    const tree = renderer
      .create(
        <Card
          height={"140px"}
          width={"100px"}
          handleClick={() => console.log("handleCick")}
          flipped={true}
          disabled={false}
          title={"Testing Card"}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
