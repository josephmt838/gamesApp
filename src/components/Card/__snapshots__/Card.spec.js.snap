// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Renders Card component 1`] = `
<div
  className="card flipped"
  onClick={[Function]}
>
  <div
    className="flipper"
  >
    <img
      alt="card"
      style={
        Object {
          "height": "140px",
          "width": "100px",
        }
      }
    />
    <h3>
      Testing Card
    </h3>
  </div>
</div>
`;
